ibm/service/cis/resource_ibm_cis_mtls.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	if host_val, ok := d.GetOk(cisMtlsHostName); ok {
+		options.SetAssociatedHostnames([]string{host_val.(string)})
+	}
+
+	result, resp, err := sess.CreateAccessCertificate(options)
+	if err != nil || result == nil {
+		return fmt.Errorf("[ERROR] Error creating MTLS access certificate %v", resp)
+	}
+
+	d.SetId(flex.ConvertCisToTfThreeVar(*result.Result.ID, zoneID, crn))
+	return ResourceIBMCISMtlsRead(d, meta)
+
+}
+
+func ResourceIBMCISMtlsRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	certID, zoneID, crn, _ := flex.ConvertTfToCisThreeVar(d.Id())
+	getOptions := sess.NewGetAccessCertificateOptions(zoneID, certID)
+	result, response, err := sess.GetAccessCertificate(getOptions)
+
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("[ERROR] Error While reading MTLS access certificate %v:%v:%v", err, response, result)
+	}
+
+	d.Set(cisID, crn)
+	d.Set(cisDomainID, zoneID)
+	d.Set(cisMtlsID, *result.Result.ID)
+	d.Set(cisMtlsCertCreatedAt, *result.Result.CreatedAt)
+	d.Set(cisMtlsCertUpdatedAt, *result.Result.UpdatedAt)
+	d.Set(cisMtlsCertExpireOn, *result.Result.ExpiresOn)
+
+	return nil
+}
+
+func ResourceIBMCISMtlsUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	certID, zoneID, _, _ := flex.ConvertTfToCisThreeVar(d.Id())
+
+	if d.HasChange(cisMtlsCertName) ||
+		d.HasChange(cisMtlsHostName) {
+
+		updateOption := sess.NewUpdateAccessCertificateOptions(zoneID, certID)
+		if host_val, ok := d.GetOk(cisMtlsHostName); ok {
+			updateOption.SetAssociatedHostnames([]string{host_val.(string)})
+		}
+
+		if name, ok := d.GetOk(cisMtlsCertName); ok {
+			updateOption.SetName(name.(string))
+		}
+
+		updateResult, updateResp, updateErr := sess.UpdateAccessCertificate(updateOption)
+		if updateErr != nil {
+			if updateResp != nil && updateResp.StatusCode == 404 {
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("[ERROR] Error while updating the MTLS cert options %v", updateResult)
+		}
+	}
+
+	return ResourceIBMCISMtlsRead(d, meta)
+}
+
+func ResourceIBMCISMtlsDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	zoneID := d.Get(cisDomainID).(string)
+
+	listOpt := sess.NewListAccessCertificatesOptions(zoneID)
```

Comment:
Do we need to loop here can't we use the first part of ID to get the instance and delete

Reply:
Yes, We need to use loop here.